pkg/cloudprovider/vsphereparavirtual/loadbalancer_test.go

```diff
@@ -44,385 +40,310 @@ import (
-					vms := &vmopv1alpha1.VirtualMachineService{
-						Status: vmopv1alpha1.VirtualMachineServiceStatus{
-							LoadBalancer: vmopv1alpha1.LoadBalancerStatus{
-								Ingress: []v1alpha1.LoadBalancerIngress{
-									{
-										IP: "10.10.10.10",
-									},
-								},
-							},
-						},
-						ObjectMeta: metav1.ObjectMeta{
-							Name: "test-vm-service-name",
-							OwnerReferences: []metav1.OwnerReference{
-								testOwnerReference,
-							},
-						},
-						Spec: vmopv1alpha1.VirtualMachineServiceSpec{
-							Type: vmopv1alpha1.VirtualMachineServiceTypeLoadBalancer,
-							Ports: []v1alpha1.VirtualMachineServicePort{
-								{
-									Name:       "test-port",
-									Port:       80,
-									TargetPort: 30800,
-									Protocol:   "TCP",
-								},
-							},
-							Selector: map[string]string{
-								vmservice.ClusterSelectorKey: testClustername,
-								vmservice.NodeSelectorKey:    vmservice.NodeRole,
-							},
-						},
-					}
-					vms.DeepCopyInto(obj.(*vmopv1alpha1.VirtualMachineService))
-					return nil
-				}
-			})
-			AfterEach(func() {
-				err := lb.EnsureLoadBalancerDeleted(context.Background(), testClustername, testK8sService)
-				Expect(err).ShouldNot(HaveOccurred())
-			})
-			It("should not return error", func() {
-				Expect(ensureErr).ShouldNot(HaveOccurred())
-				Expect(status.Ingress[0].IP).Should(Equal("10.10.10.10"))
-			})
-		})
+	// should pass without error
+	err := lb.EnsureLoadBalancerDeleted(context.Background(), testClustername, testK8sService)
+	assert.NoError(t, err)
 
-		Context("when delete loadbalancer", func() {
-			BeforeEach(func() {
-				testK8sService = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      testK8sServiceName,
-						Namespace: testK8sServiceNameSpace,
-					},
-				}
-			})
-			It("should pass without error", func() {
-				err := lb.EnsureLoadBalancerDeleted(context.Background(), testClustername, testK8sService)
-				Expect(err).ShouldNot(HaveOccurred())
-			})
-			It("should ignore not found error", func() {
-				fcw.DeleteFunc = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
-					return apierrors.NewNotFound(v1alpha1.Resource("virtualmachineservice"), testClustername)
-				}
-				err := lb.EnsureLoadBalancerDeleted(context.Background(), "test", testK8sService)
-				Expect(err).ShouldNot(HaveOccurred())
-			})
-			It("should return error", func() {
-				fcw.DeleteFunc = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
-					return fmt.Errorf("an error occurred while deleting load balancer")
-				}
-				err := lb.EnsureLoadBalancerDeleted(context.Background(), "test", testK8sService)
-				Expect(err).To(MatchError("an error occurred while deleting load balancer"))
-			})
-		})
-	})
-})
+	// should ignore not found error
+	fcw.DeleteFunc = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
```

Comment:
Nit: would be best if could be several test cases with different DeletFunc

Reply:
got it, yeah I agree